js/src/common/utils/ItemList.ts

```diff
@@ -10,23 +14,15 @@ class Item {
  * by priority.
  */
 export default class ItemList {
-  constructor() {
-    /**
-     * The items in the list.
-     *
-     * @type {Object}
-     * @public
-     */
-    this.items = {};
-  }
+  /**
+   * The items in the list
+   */
+  items: { [x: string]: Item } = {};
```

Comment:
I assume the name `x` is a required part of this syntax, which I assume to mean that all keys in the object need to be of type `string` and mapping to a value of `Item`?

Reply:
Yep. That `x` could be `key` instead.

Reply:
Indeed, [the name is only for readability](https://basarat.gitbook.io/typescript/type-system/index-signatures#declaring-an-index-signature). I'm changing it to `key` now.